src/tiled/replaceobjectstile.cpp

```diff
@@ -0,0 +1,51 @@
+#include "replaceobjectstile.h"
+
+#include "mapdocument.h"
+#include "mapobjectmodel.h"
+
+#include <QCoreApplication>
+
+namespace Tiled {
+namespace Internal {
+
+ReplaceObjectsTile::ReplaceObjectsTile(MapDocument *mapDocument,
+                                       const QList<MapObject *> &mapObjects,
+                                       Tile *tile)
+    : QUndoCommand(QCoreApplication::translate("Undo Commands",
+                                               "Replace %n Object/s Tile",
+                                               nullptr, mapObjects.size()))
+    , mMapDocument(mapDocument)
+    , mMapObjects(mapObjects)
+    , mTile(tile)
+{}
+
+void ReplaceObjectsTile::replace()
+{
+    Tile * const tile = mTile;
+
+    mOriginalTiles.clear();
+
+    for (MapObject *object : mMapObjects) {
+        mOriginalTiles.append(object->cell().tile());
+        swapObjectTile(object, tile);
+    }
+
+    emit mMapDocument->mapObjectModel()->objectsChanged(mMapObjects);
+}
+
+void ReplaceObjectsTile::restore()
+{
+    for (int i = 0; i < mMapObjects.size(); ++i)
+        swapObjectTile(mMapObjects[i], mOriginalTiles[i]);
+
+    emit mMapDocument->mapObjectModel()->objectsChanged(mMapObjects);
+}
+
+void ReplaceObjectsTile::swapObjectTile(MapObject *object, Tile *tile) {
```

Comment:
Coding style: `{` should be on its own line for functions.

Naming: this function should just be called `setObjectTile`.

Also, this function can be a local static because it doesn't use any members of `ReplaceObjectsTile` and it's not used from elsewhere.